lib/simplecov-lcov.rb

```diff
@@ -12,53 +13,59 @@ class LcovFormatter
       # _result_ :: [SimpleCov::Result] abcoverage result instance.
       def format(result)
         create_output_directory!
-        if self.class.report_with_single_file?
+
+        if report_with_single_file?
           write_lcov_to_single_file!(result.files)
         else
           result.files.each { |file| write_lcov!(file) }
         end
 
-        puts "Lcov style coverage report generated for #{result.command_name} to #{SimpleCov::Formatter::LcovFormatter.output_directory}."
+        puts "Lcov style coverage report generated for #{result.command_name} to #{lcov_results_path}"
       end
 
       class << self
-        attr_writer :report_with_single_file
-
-        def report_with_single_file?
-          !!@report_with_single_file
-        end
-
-        # Output directory for generated files.
-        # ==== Return
-        # Path for output directory.
-        def output_directory
-          File.join(SimpleCov.coverage_path, 'lcov')
+        def config
+          @config ||= SimpleCovLcov::Configuration.new
+          yield @config if block_given?
+          @config
         end
 
-        # Output path for single file report.
-        # ==== Return
-        # Path for output path of single file report.
-        def single_report_path
-          basename = Pathname.new(SimpleCov.root).basename.to_s
-          File.join(output_directory, "#{basename}.lcov")
+        def report_with_single_file=(value)
+          config.report_with_single_file = value
```

Comment:
This method is left only to support older versions. I would like to deprecate it. @fortissimo1997, what is your opinion on this?

Reply:
@povilasjurcys 
This method needs to be deprecated, I think.
If so, next version has to be 1.0.0!

Reply:
I added deprecation warning for now